### ISSUE TITLE
Do not use the term `whitelist`, `blacklist`

### ISSUE THREAD

**Issue by whitney-cumber** [neomutt/neomutt#3619]
This is obviously a political matter.

Avoid the terms whitelist and blacklist because of their connotation.
The biggest impact would be renaming `sidebar_whitelist` into something else, e.g. `sidebar_permanent_mailboxes` (+ the synonym for backwards compatibility). All other places (~20) are code comments.

If I get a go ahead I would rename those.

Cons: maintenance burden of the synonym.

**Comment by gahr** [comment]
If we go with this, I think we should accept the most accepted / widespread alternative, which - AFAIK, and please correct me if I'm wrong - would be `allow_list` and `block_list`.

**Comment by whitney-cumber** [comment]
A agree that allow_list, block_list are probably the most used replacements but that's because they are actually used in the context of blocking/allowing, e.g. access lists. For the sidebar, however, the mailboxes are not "blocked" from being on the list nor are they allowed (all mailboxes are allowed on the sidebar as long as they have new mail). Generally, one should replace "blacklist"/"whitelist" by the verb/noun they actually are used for.

That's why I suggested "permanent" in this context.  We can also use "unhide", "fixed", "always_show", ... but now we are drifting into the realm of bikeshedding, which is a waste of time until we (which is more like a you) have decided about the "if we do this" part.

**Comment by flatcap** [comment]
Yes, I think we should get rid of the term.

> For the sidebar, however, the mailboxes are not "blocked"

Indeed.  We only need the inclusive side.

I quite like: **`sidebar_always_show`** -- it's clear what it's going to do.
(We could add `_mailboxes` to the end, but that gets a bit verbose)

Please can you make change.
Thanks!

**Comment by flatcap** [comment]
Whilst googling the subject, I found this, which nicely summarises the options:

![lists](https://cdn.ttgtmedia.com/rms/onlineimages/sap-sap_name_changes-f.png)
https://cdn.ttgtmedia.com/rms/onlineimages/sap-sap_name_changes-f.png

**Comment by whitney-cumber** [comment]
Ok, let the bikeshedding begin!

Currently, I'm undecided between `sidebar_always_show` and `sidebar_always_display`.  Any preferences?

**Comment by gahr** [comment]
My take is `sidebar_stick` and `sidebar_unstick` or any verb which is better than `stick` in this context. The rationale is that these are commands, so I like to read them as "verb object". Also, the `un` would go with the verb.

**Comment by mschilli87** [comment]
@gahr: show/hide? :shrug:

**Comment by flatcap** [comment]
Hmm... it's the `mailboxes` command that actually causes the folder to be shown,
and config like `sidebar_new_mail_only` that might cause it to be hidden.

**Comment by gahr** [comment]
Yeah it’s not about showing / hiding, it’s about keeping them displayed/hidden even though other configs would imply otherwise

**Comment by mschilli87** [comment]
Then I'd 2nd `always_show`/`always_hide`.

**Comment by auouymous** [comment]
> My take is sidebar_stick and sidebar_unstick or any verb which is better than stick in this context.

I don't know what the feature does, but would `sidebar_sticky` and `sidebar_unsticky` be better than `sidebar_stick` and `sidebar_unstick`?

**Comment by auouymous** [comment]
Ah, so maybe `sidebar_set_(un)sticky` or `sidebar_(un)persist`. Should the command have a `_folder` suffix to help indicate what it does?

**Comment by flatcap** [comment]
Yes, `sidebar_set_sticky` would be a good alternative.

> Should the command have a `_folder`

Possibly... probably...
We don't want the commands becoming too long and painful to use :-)

**Comment by flatcap** [comment]
One last update from IRC,
@jindraj suggested `sidebar_(un)pin` which got a thumbs up from @gahr.

**Comment by whitney-cumber** [comment]
To summarise the suggestions so far (in chronological order):

* `(un)sidebar_always_show`    (+2)
* `(un)sidebar_always_display` (+1)
* `sidebar_(un)stick`          (+1)
* `sidebar_set_(un)sticky`     (+1)
* `sidebar_(un)pin`            (+3)

[the number is the time a supporting comment was made; but no indication of its overall acceptance]

---

Personally, I favour one of `sidebar_pin`, `sidebar_always_show`, `sidebar_always_display` probably with pinning slightly ahead.

---

**Now to something orthogonal**

Since this  was implicitly mentioned/assumed during this discussion, I'd like to share my thoughts on the following question:

   *How to name the "reverse" command?*

There are two canonical candidates, namely:

* `unsidebar_foo`
* `sidebar_unfoo`

---

The arguments for the two choices, as I see, are:

`unsidebar_foo`
* coherent naming scheme: all other commands are also of the form `bar`/`unbar`.

`sidebar_unfoo`
* a more natural wording, compare: `unsidebar_pin` vs. `sidebar_unpin`.


Skimming through NeoMutt's other commands, I noticed they are mostly verbs, e.g. `subscribe`, `ignore`, `bind`,  but not all, e.g. `mailto_allow`.  The latter example would also read more natural with `mailto_unallow`.  It seems also worth noting that `sidebar_whitelist` seems to be the only command which is of the form `<prefix>_<real-command-name>`, which might warrant a different treatment.

Nevertheless, I would opt for `unsidebar_foo` simply to not break the user expectation "if I add `un` I revert the meaning".
Even if that means that the resulting word/command is of questionable linguistic elegance.

**Comment by mschilli87** [comment]
Why make `sidebar` a prefix? Is `[un]pin` sufficient? Would `[un]pin_[in_]sidebar` be too long?

**Comment by auouymous** [comment]
`sidebar_(un)pin` is better than anything I suggested.

I don't agree with `unsidebar_pin` as it makes no sense.

**Comment by flatcap** [comment]
# Results

We will rename:
- **`sidebar_whitelist`** to **`sidebar_pin`**
- **`unsidebar_whitelist`** to **`sidebar_unpin`**

---

Thanks to everyone for their input :heart:

**Comment by whitney-cumber** [comment]
> There **is** a precedent for asymmetric commands: imap has: `subscribe-to` and `unsubscribe-from`.

Ok, let me rephrase my statement: all other *documented* commands are of the form `foo`/`unfoo`. (#3654)
That said if we still want to deviate from this convention, I'll rename them to `sidebar_pin`, `sidebar_unpin`.

**Comment by gahr** [comment]
Most (all?) other commands don't have a prefix :)

**Comment by flatcap** [comment]
> I'll rename them to `sidebar_pin`, `sidebar_unpin`.

Great, thanks!
